### PR TITLE
[Revisions] High level interface: Views and Controllers

### PR DESCRIPTION
**Pull request by kshala-ford**
This PR contains revisions requested in #511 (sorry for the weird PR pointing to wrong ocmmits but the files changed should be correct).

### CODE REVIEW

**Comment by kshala-ford** [comment]
@theresalech The PR containing revisions is ready for merge.

File: proposals/0176-high-level-interface-views-and-controllers.md [hunk 1]
```diff
@@ -113,40 +121,78 @@ The display layout that should be used while the view controller is presented. I
 
 In order to support non-predefined layouts (OEM or custom layouts) the app developer can set the name of the custom template.
 
-#### `showViewController`
+#### `manager`
 
-Similar to [`UIViewController.showViewController:sender:`](https://developer.apple.com/documentation/uikit/uiviewcontroller/1621377-showviewcontroller?changes=_2&language=objc) This method should allow app developer to present another view controller in the primary context. The method is a shortcut to the shared application and the view controller manager. Depending on the manager's mode the method either replaces the current presented view controller or pushes it to the view controller stack (see `SDLModalViewController`).
+A weak reference to the view controller manager object managing the current view controller. It's a convenient property for easy access to present antoher view controller.
```
Comment (theresalech) [inline proposals/0176-high-level-interface-views-and-controllers.md]: Please fix typo "another"

File: proposals/0176-high-level-interface-views-and-controllers.md [hunk 2]
```diff
@@ -113,40 +121,78 @@ The display layout that should be used while the view controller is presented. I
...
+
+This array is used as a stack for view controllers pushed by the app developer. It can be accessed and replaced by the app developer at any time.
+
+#### `topViewController`
+
+The top view controller is the most top from the stack presented on the head unit. This property is pointing to the last view controller of the stack.
+
+#### `rootViewController` and `setRootViewController`
+
+The very first view controller of the stack is returned by `rootViewController`. Settings a new root view controller will clear the stack and set a new root.
```
Comment (theresalech) [inline proposals/0176-high-level-interface-views-and-controllers.md]: Please update the second sentence to "Setting a new..."

File: proposals/0176-high-level-interface-views-and-controllers.md [hunk 3]
```diff
@@ -203,12 +249,23 @@ The image view will be used for the primary and secondary graphic in the order a
 
 #### SDLButtonView
 
-Every button view added to the view controller's view will be used for the button managers `softButtons` array. If desired the app developer can manage the soft button views in a dedicated subview of type `SDLView`.
+Every button view added to the view controller's view will be used for the button managers `softButtons` array. If desired the app developer can manage the soft button views in a dedicated subview of type `SDLView`. This view will maintain a private reference to an `SDLSoftButtonObject` object wrapping and bypassing all the initializers, properties and methods. Below is a preview of the `SDLButtonView` class for illustration. The implementation will incude *all* the methods etc.
```
Comment (theresalech) [inline proposals/0176-high-level-interface-views-and-controllers.md]: Please fix typo in last sentence "include"

File: proposals/0176-high-level-interface-views-and-controllers.md [hunk 4]
```diff
@@ -70,19 +70,11 @@ This method is called after `viewWillDisappear`.
 
 ### SDLViewController
 
-The view controller class is called `SDLViewController`. It implements the view controller and provides the necessary classes for the lifecycle. It's expected that the app developer creates subclasses of this base class.
+The view controller class is called `SDLViewController`. It's expected that the app developer creates subclasses of this base class. The methods for the lifecycle are realized using a protocol with optional methods. This way app developer can implement the methods in their subclasses without the need of calling super. `SDLViewController` implements the delegate but not the methods except `loadView`. The color scheme can be explicitly set presenting a view controller. When pushing another view controller to the stack it'll inherit the color scheme if not explicitly set.
```
Comment (theresalech) [inline proposals/0176-high-level-interface-views-and-controllers.md]: Please update the fourth sentence to "This way the app developer..." (adding "the").

**Comment by kshala-ford** [comment]
@theresalech thank you. All mistakes should be fixed now.

File: proposals/0176-high-level-interface-views-and-controllers.md [hunk 5]
```diff
@@ -176,7 +176,7 @@ The top view controller is the most top from the stack presented on the head uni
 
 #### `rootViewController` and `setRootViewController`
 
-The very first view controller of the stack is returned by `rootViewController`. Settings a new root view controller will clear the stack and set a new root.
+The very first view controller of the stack is returned by `rootViewController`. Settings a new root view controller will clear the stack and setting a new root.
```
Comment (theresalech) [inline proposals/0176-high-level-interface-views-and-controllers.md]: The last sentence should read "Setting a new root view controller will clear the stack and set a new root."

**Comment by kshala-ford** [comment]
@theresalech Whoops... I guess misunderstood your first comment... 

Setting set and settings should now be all set...

**Pull request merged**